storage-client/src/client.rs

```diff
@@ -36,11 +45,147 @@ impl StorageClientImpl {
 
 #[async_trait::async_trait]
 impl StorageClient for StorageClientImpl {
+    async fn get_info_from_catalog(&self, _request: StorageRequest) -> Result<RequestParams> {
+        // FIXME (kunle): Need to discuss with the catalog team.
+        // The following line serves as a placeholder for now.
+        Ok(RequestParams::File("userdata1.parquet".to_string()))
+    }
+
     async fn request_data(&self, _request: StorageRequest) -> Result<Receiver<RecordBatch>> {
-        todo!()
+        // First we need to get the location of the parquet file from the catalog server.
+        let _location = match self.get_info_from_catalog(_request).await? {
+            RequestParams::File(location) => location,
+            _ => {
+                return Err(anyhow!(
+                    "Failed to get location of the file from the catalog server."
+                ));
+            }
+        };
+
+        // Then we need to send the request to the storage server.
+        let url = format!("{}file/{}", self._storage_server_endpoint, _location);
+        let client = Client::new();
+        println!("Requesting data from: {}", url);
+        let response = client.get(url).send().await?;
+        if response.status().is_success() {
+            // Store the streamed Parquet file in a temporary file.
+            // FIXME: 1. Do we really need streaming here?
+            //       2. Do we need to store the file in a temporary file?
+            let temp_dir = tempdir()?;
+            let file_path = temp_dir.path().join("tmp.parquet");
+            let mut file = File::create(&file_path)?;
+            let mut stream = response.bytes_stream();
+            while let Some(chunk) = stream.next().await {
+                let chunk = chunk?;
+                file.write_all(&chunk)?;
+            }
+
+            // Convert the Parquet file to a record batch.
+            let file = AsyncFile::open(file_path).await.unwrap();
+            let builder = ParquetRecordBatchStreamBuilder::new(file)
+                .await
+                .unwrap()
+                .with_batch_size(3);
+
+            let mask = ProjectionMask::all();
+            let stream = builder.with_projection(mask).build().unwrap();
+            let results = stream.try_collect::<Vec<_>>().await.unwrap();
+
+            // Return the record batch as a stream.
+            let (tx, rx) = channel(1);
```

Comment:
We may extract this into a const variable. BTW I think we can have a larger buffer size here for more flexible message traffic.